### PR TITLE
fix: add logic handling zero strike or low strike scenarios

### PR DESCRIPTION
**Pull request by antoncoding**
### New Restriction on SVI calculation
1. cap `w` at `MAX_VAR` before calculating vol
2. cap `k` between range [-B, B], where  `B = SCALAR x sqrt(a + b*sig)` (min_k = -B, max_k = B)

### Special Case
* if strike is 0 => return 0
* if strike is close to 0 (making strike / fwd 0 in solidity), assign k = min_k, (-B) from above. This avoid Overflow error from `FixPointMath.ln`

## Testing
* `testFuzzGetVolCapped` show that with a example set of svi parameters, (a SVI parameter is provided by vol feed on per-expiry bases), no matter how dramatic the strike & fwdPrice moves, the vol returned will not be easily manipulated, it will always be bonded within a reasonable range.

### CODE REVIEW

File: src/math/SVI.sol [hunk 1]
```diff
@@ -5,6 +5,8 @@ pragma solidity ^0.8.0;
 import "src/decimals/SignedDecimalMath.sol";
 import "src/decimals/DecimalMath.sol";
 import "./FixedPointMathLib.sol";
+import "./IntLib.sol";
```
Comment (0xdomrom) [inline src/math/SVI.sol:8]: Is this used anywhere?

File: src/math/SVI.sol [hunk 2]
```diff
@@ -56,9 +69,35 @@ library SVI {
...
+    if (sk == 0) {
+      k = -k_bound;
+    } else {
+      // k = ln (strike / fwd)
+      k = FixedPointMathLib.ln(sk);
+      // make sure -B < k < B
+      if (k > k_bound) {
+        k = k_bound;
+      } else if (k < -k_bound) {
+        k = -k_bound;
```
Comment (0xdomrom) [inline src/math/SVI.sol:99]: would do explicit returns here (i.e. `return -k_bound;`) to exit early, a bit cleaner imo

**Review by 0xdomrom — APPROVED** [review]
lgtm!

File: src/math/SVI.sol (hunk 2 above)
Comment (antoncoding) [reply src/math/SVI.sol]: nice suggestion, i was previously calculating k in the `getVol` function so has to do assigning. we can actually clean it up by quite a lot by returning early in the function

File: src/math/SVI.sol (hunk 1 above)
Comment (antoncoding) [reply src/math/SVI.sol]: great catch

**Pull request merged**